src/fleet/azext_fleet/custom.py

```diff
@@ -212,6 +216,89 @@ def get_credentials(cmd,  # pylint: disable=unused-argument
         raise CLIError("Fail to find kubeconfig file.")
 
 
+def assign_fleet_identity(cmd,
+                 client,
+                 resource_group_name,
+                 name,
+                 system_assigned=None,
+                 user_assigned=None,
+                 no_wait=False):
+    fleet_patch_model = cmd.get_models(
+        "FleetPatch",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    fleet_managed_service_identity_model = cmd.get_models(
+        "ManagedServiceIdentity",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+
+    # Handle all 6 permutations of mi_system & mi_user assigned parameters
+    if system_assigned is True and user_assigned is not None:
+        raise CLIError("A fleet can either have a system or user assigned identity, not both.")
+    if system_assigned is None and user_assigned is None:
+        managed_service_identity = None
+    elif system_assigned is False and user_assigned is None:
+        managed_service_identity = fleet_managed_service_identity_model(type="None")
+    elif system_assigned is True and user_assigned is None:
+        managed_service_identity = fleet_managed_service_identity_model(type="SystemAssigned")
+    else:
+        managed_service_identity = fleet_managed_service_identity_model(type="UserAssigned")
+        user_assigned_identity_model = cmd.get_models(
+                "UserAssignedIdentity",
+                resource_type=CUSTOM_MGMT_FLEET,
+                operation_group="fleets"
+            )
+        managed_service_identity.user_assigned_identities = {user_assigned: user_assigned_identity_model()}
+
+    fleet_patch = fleet_patch_model(
+        identity=managed_service_identity
+    )
+
+    return sdk_no_wait(no_wait, client.begin_update, resource_group_name, name, fleet_patch, polling_interval=5)
+
+
+def remove_fleet_identity(cmd,
+                 client,
+                 resource_group_name,
+                 name,
+                 system_assigned=False,
+                 user_assigned=False,
+                 no_wait=False):
+    fleet_patch_model = cmd.get_models(
+        "FleetPatch",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    fleet_managed_service_identity_model = cmd.get_models(
+        "ManagedServiceIdentity",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+
+    if system_assigned is False and user_assigned is False:
+        raise CLIError("Either --system-assigned is set to true, or pass an empty --user-assigned")
```

Comment:
Unfortunately, I think you need to go an extra mile here:

if --system-assigned is set, you should get the fleet, validate it's system assigned, then set it to None
if --user-assigned is set, you should get the fleet, validate it's user assigned with the identity provided, then set it to None

e.g.:
- remove --user-assigned should **fail** to set a system-assigned to None
- remove --user-assigned foo should **fail** to set a user-assigned bar to None

Reply:
Updated to get and check.